src/map.cpp

```diff
@@ -277,7 +277,7 @@ std::unique_ptr<vehicle> map::detach_vehicle( vehicle *veh )
         return std::unique_ptr<vehicle>();
     }
 
-    if( veh->smz < -OVERMAP_DEPTH && veh->smz > OVERMAP_HEIGHT ) {
+    if( veh->smz < -OVERMAP_DEPTH || veh->smz > OVERMAP_HEIGHT ) {
```

Comment:
That was an obvious error in out-of-bounds check - vehicle cannot be above `OVERMAP_HEIGHT`  and below `OVERMAP_DEPTH` at he same time, so `OR (||)` should be used instead of `AND (&&)`.